src/js/tour.js

```diff
@@ -129,17 +130,26 @@ export class Tour extends Evented {
   /**
    * Calls _done() triggering the 'cancel' event
    * If `confirmCancel` is true, will show a window.confirm before cancelling
+   * If `confirmCancel` is a function, will call it and wait for the return value,
+   * and only cancell when the value returned is true
    */
-  cancel() {
-    if (this.options.confirmCancel) {
-      const cancelMessage =
-        this.options.confirmCancelMessage ||
-        'Are you sure you want to stop the tour?';
-      const stopTour = window.confirm(cancelMessage);
+  async cancel() {
+    if (!this.options.confirmCancel) {
+      this._done('cancel');
+      return;
+    }
+    const type = Object.prototype.toString.call(this.options.confirmCancel);
+    if (type === '[object Function]' || type === '[object AsyncFunction]') {
```

Comment:
```suggestion
    if (typeof this.options?.confirmCancel === 'function') {
```

Would this work? Seems a bit simpler.

Reply:
Thanks, this does make it easier, which I will adopt later, but most importantly, based on the error message reported by the checker as shown below, the root of the error may not be here?

![image](https://user-images.githubusercontent.com/49258735/213844283-69d877a9-bfab-4d33-9564-6ea978ace9f1.png)

Of course, it could also be that find this problem here and then it just quits